qiskit/qpy/binary_io/circuits.py

```diff
@@ -702,7 +805,12 @@ def _write_instruction(file_obj, instruction, custom_operations, index_map, use_
     file_obj.write(gate_class_name)
     file_obj.write(label_raw)
     if condition_type is type_keys.Condition.EXPRESSION:
-        value.write_value(file_obj, op_condition, index_map=index_map)
+        value.write_value(
+            file_obj,
+            op_condition,
+            index_map=index_map,
+            standalone_var_indices=standalone_var_indices,
+        )
```

Comment:
Not in the writers, for sure - the variable is always defined, it just defaults to `None` or empty-dict.  When I add the backwards-compat tests and the minimum-version test, I can check end-to-end.

Reply:
Oh no wait, I just understood you meant it in the sense of "in order to _throw_ an error".  In that case, also no, I'm fairly sure - we'd already have errored out when writing the circuit header, since the circuit context knows about all the standalone variables that might be directly used by its instructions.